### PR TITLE
(DEV-3685, DEV-3686) Adds "Add a Prayer" UI

### PR DESCRIPTION
**Pull request by redreceipt**
Screen Shot 2019-04-15 at 3 47 16 PM

### What does this PR do, or why is it needed?

This will add the UI to add a new prayer.

### How do I test this PR?

`yarn storybook` and `yarn test`

---

> I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))

## TODO

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] Closes [DEV-3685, DEV-3686]
- [x] No new warnings in tests, in storybook, and in-app
- [x] Upload GIF(s) of iOS and Android if applicable
- [x] Set two relevant reviewers

## REVIEW

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
@IsaacHardy any idea what this build error is saying:

Screen Shot 2019-04-08 at 12 17 02 PM

**Comment by cdanwards** [comment]
@redreceipt did you get your question answered?

**Comment by richarddubay** [comment]
@redreceipt Maybe I didn't hear this conversation, but I can run your branch without error. Where is the error happening? I don't see anything at all in Travis for this PR.

**Comment by IsaacHardy** [comment]
I thought this was the error I helped you with two days ago?

**Comment by redreceipt** [comment]
Yes I got it up and running, made good progress today.

File: packages/apollos-data-prayer/package.json [hunk 1]
```diff
@@ -39,4 +39,4 @@
   "publishConfig": {
     "access": "public"
   }
-}
\ No newline at end of file
+}
```
Comment (IsaacHardy) [inline packages/apollos-data-prayer/package.json:42]: uhh what github? lol

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import PropTypes from 'prop-types';
+import { H5, styled, Button, Card, CardContent } from '@apollosproject/ui-kit';
+import AddPrayerHeader from '../AddPrayerHeader';
+
+const StyledCardContent = styled(() => ({
+  alignItems: 'center',
+}))(CardContent);
+
+const Description = styled(({ theme }) => ({
+  color: theme.colors.text.secondary,
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js]: I think you can remove this. The text should just be normal colored.
```suggestion
```

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  marginBottom: theme.sizing.baseUnit,
+}))(H5);
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js]: Also, this can just be `<BodyText>`
```suggestion
}))(BodyText);
```

**Review by IsaacHardy — COMMENTED** [review]
@redreceipt I am unable to test this with storybook. Looks like there is an error 
Screen Shot 2019-04-18 at 1 28 34 PM

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  width: '100%',
+}))(Button);
+
+const ButtonLabel = styled(({ theme }) => ({
+  color: theme.colors.text.primary,
+}))(H5);
+
+const AddPrayerCard = memo(({ imgSrc, description }) => (
+  <Card>
+    <StyledCardContent>
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js:29]: Could probably add a `<PaddedView>` in between `<Card>` and `<StyledContentCard>` but honestly, I'd like to see us modularize this since there are at least 3 different places where we use very similar UI. I'm keeping a list of polish items ... I just need to make a task to list them all.

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  alignItems: 'center',
+}))(CardContent);
+
+const Description = styled(({ theme }) => ({
+  color: theme.colors.text.secondary,
+  marginBottom: theme.sizing.baseUnit,
+}))(H5);
+
+const AddPrayerButton = styled(({ theme }) => ({
+  marginTop: theme.sizing.baseUnit * 2,
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js:19]: Between the `marginTop` here and the `marginBottom` on the description I think we've got a little too much white space. I think just `baseUnit` here would be fine.
```suggestion
  marginTop: theme.sizing.baseUnit,
```

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.stories.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.stories.js:1]: I got a lot of warnings when I clicked on the first story. Is this all from Formik?

Screenshot 2019-04-18 13 34 54

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+              editable
+              placeholder="Start typing your prayer..."
+              onChangeText={handleChange('prayer')}
+              onBlur={handleBlur('prayer')}
+              value={values.prayer}
+            />
+          </PaddedView>
+          <BottomView>
+            <PaddedView>
+              <Switch
```
Comment (richarddubay) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js:70]: Looking at the mock below, I'd love to see this Anonymous switch look more like that _if it's possible_. If not, ignore.

Screenshot 2019-04-18 13 39 44

File: packages/apollos-data-prayer/package.json (hunk 1 above)
Comment (redreceipt) [reply packages/apollos-data-prayer/package.json:42]: can you get storybook to run on master?

**Review by richarddubay — CHANGES_REQUESTED** [review]

File: packages/apollos-data-prayer/package.json (hunk 1 above)
Comment (redreceipt) [reply packages/apollos-data-prayer/package.json:42]: that's an error in your node modules, did you `yarn`?

Comment (IsaacHardy) [reply packages/apollos-data-prayer/package.json:42]: PS: This isn't the error image but I'm happy to use this as our convo piece.

Yes. I did yarn.

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.stories.js (hunk 6 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.stories.js:1]: yes that's all from formik but if you yarn'd you should have gotten the updated `formik` that gets rid of those warnings

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js (hunk 7 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js:70]: I could put it a core PR to swap the side the label is on for the `ui-kit/Switch`?

File: packages/apollos-data-prayer/package.json (hunk 1 above)
Comment (redreceipt) [reply packages/apollos-data-prayer/package.json:42]: I'm confused. What error are you getting?

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js (hunk 7 above)
Comment (richarddubay) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js:70]: I don't think the sides the things are on are that big of a deal. What if you just centered it? Is that possible?

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.stories.js (hunk 6 above)
Comment (richarddubay) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.stories.js:1]: This is after I yarn'd. So ... closer. Would this be need an update to `Switch` then? I'm guessing yes and if so, we could probably just ignore it for now. Although this might be worth fixing in TAP core so that they go away ... sometime down the road. 

Screenshot 2019-04-18 14 09 26

Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/AddPrayerForm.stories.js:1]: ...and yes I have an open core PR to fix the `Switch` :)

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js (hunk 4 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerCard/index.js:29]: love that! I'll leave this here and let you polish it all up later, make sure to add to that list "organize prayer stuff"

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js (hunk 7 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js:70]: Screen Shot 2019-04-18 at 2 45 13 PM
 what do you think?

**Review by richarddubay — DISMISSED** [review]
Looks good. We'll catch up some style tweaks on a polish pass. I can't wait to see this functioning. 🎉

**Review by richarddubay — DISMISSED** [review]
:shipit:

**Comment by redreceipt** [comment]
@IsaacHardy can you give this another pass, it's working for rich and it's blocking some other work for me

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+import PropTypes from 'prop-types';
+
+import { Avatar, styled, H3 } from '@apollosproject/ui-kit';
+
+const HeaderContent = styled(() => ({
+  alignItems: 'center',
+}))(View);
+
+const Title = styled(({ theme }) => ({
+  marginTop: theme.sizing.baseUnit * 0.5,
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js:12]: We shouldn't need to add margin/padding to typography components.

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+  marginBottom: theme.sizing.baseUnit * 2,
+}))(H3);
+
+const AddPrayerHeader = memo(({ imgSrc, title }) => (
+  <HeaderContent>
+    <Avatar source={imgSrc} />
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js]: the `<Avatar/>` component defaults to `size={'small'}`. But that doesn't seem to be the size we need. Try `size={'medium or large'}`

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    onSubmit={(values) => console.log(values)}
+  >
+    {({ handleChange, handleBlur, handleSubmit, values }) => (
+      <ModalView {...otherProps}>
+        <FlexedView>
+          <AddPrayerHeader imgSrc={imgSrc} />
+          {/* TODO: add underline={false} prop to TextInput, 
+        pending https://github.com/ApollosProject/apollos-prototype/issues/629 */}
+          <PaddedView>
+            <TextInput
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js]: Is there anyway to make this TextInput more like a paragraph input rather than a single line input?

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+              placeholder="Start typing your prayer..."
+              onChangeText={handleChange('prayer')}
+              onBlur={handleBlur('prayer')}
+              value={values.prayer}
+            />
+          </PaddedView>
+          <BottomView>
+            <SwitchContainer>
+              <Switch
+                label="Share anonymously"
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js]: This label isn't our primary color.

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+const BottomView = styled(() => ({
+  justifyContent: 'flex-end',
+}))(FlexedView);
+
+const SwitchContainer = styled(() => ({
+  width: '60%',
+  alignSelf: 'center',
+}))(PaddedView);
+
+const AddPrayerForm = memo(({ imgSrc, btnLabel, ...otherProps }) => (
```
Comment (IsaacHardy) [inline packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js:45]: Is this form setup to update the `imgSrc` when the `<Switch/>` component is toggled? These can exist as pure components but we'll need to make sure that the parent component handles the passing down of data on toggle.

**Review by IsaacHardy — CHANGES_REQUESTED** [review]
Added some comments for changes and discussion.

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js (hunk 8 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js:12]: @IsaacHardy agreed, but how do you suggest we space out components to match the design otherwise?

This is with no margins:
header

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js (hunk 9 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js]: 👍

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js (hunk 10 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js]: @IsaacHardy Yep, totally. I think what makes it look like a single line is the underline. I have an open PR in core that gives us the option to style the `TextInput` without the underline which makes it look much more open. Do you think we can just wait on that to make it in?

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js (hunk 11 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js]: does it need to be?

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js (hunk 12 above)
Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js:45]: Not sure I understand...the `imgSrc` prop is for the avatar and that won't change based on the switch. I could make the prop name more explicit? Or maybe that's not what you're getting at.

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js (hunk 8 above)
Comment (richarddubay) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerHeader/index.js:12]: I think for the header, which this is, it's okay to just use the `H3` without the `marginTop`. In this case it's probably closer to the mock without it.

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js (hunk 11 above)
Comment (richarddubay) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js]: Is there a `color` prop on the `Switch`? If not, we'd have to bring over the whole component and edit it just to change the color. Maybe a PR to core to allow a color prop so we could use whatever color we wanted for the label in the future would be the right call here?

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js (hunk 12 above)
Comment (richarddubay) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js:45]: If I had to guess I bet what Isaac is suggesting is that if the anonymous switch is on, then the image should change to something that isn't the person's profile image. 

Screenshot 2019-04-22 14 46 58

File: packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js (hunk 10 above)
Comment (IsaacHardy) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js]: When I type into the input, it never returns to the next line but rather continues to travel horizontal. I am hoping we can create the input to function like the former.

Comment (redreceipt) [reply packages/newspringchurchapp/src/prayer/AddPrayer/AddPrayerForm/index.js]: ah, I'll work on that

**Review by IsaacHardy — APPROVED** [review]
LGTM
